### PR TITLE
[builtins][arm64] Implement __init_cpu_features_resolver on Apple platforms

### PR DESCRIPTION
**Pull request by jroelofs**
This is a re-land of https://github.com/llvm/llvm-project/pull/73685

### CODE REVIEW

File: compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  // available through libSystem, which we need anyway for the sysctl, so this
+  // does not add a new dependency.
+
+  static dispatch_once_t onceToken = 0;
+  dispatch_once(&onceToken, ^{
+    // https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics
+    static struct {
+      const char *sysctl_name;
+      enum CPUFeatures feature;
+    } features[] = {
```
Comment (ilinpv) [inline compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: Do you need support for AdvSIMD_HPFPCvt, FEAT_LSE2,  FEAT_SHA512 or for other feature in Function Multi Versioning? Specification is not final yet and open to proposals.

Comment (jroelofs) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: Sure, I can add entries for those. Looks like I missed them in my manual intersection of the [sysctl docs](https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_instruction_set_characteristics) with the `FEAT_*` table.

> Specification is not final yet and open to proposals.

Any word on when FMV might come out of beta?

Comment (jroelofs) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: Ohh, I see you're asking a different question than I thought. Yeah, those three would probably be useful. Unfortunately, looks like we're just about out of bits in `__aarch64_cpu_features.features`.

Comment (jroelofs) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: Another thing that came up when discussing the frontend attributes for these: we may want to have user-selectable overrides for the priorities for (hypothetical) cases where the prescribed numbers in the table don't match the selection behavior we need at runtime. I don't have a concrete example in mind yet.

Comment (ilinpv) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: > Looks like we're just about out of bits in `__aarch64_cpu_features.features`.
Next field features2 can be added to __aarch64_cpu_features with setting FEAT_EXT bit to 1 and changing initialization condition to if (__aarch64_cpu_features.features &  (1ULL << FEAT_INIT) )

Comment (ilinpv) [reply compiler-rt/lib/builtins/cpu_model/aarch64/fmv/apple.inc:29]: > Any word on when FMV might come out of beta?

There's no specific deadline right now, possibly within a year or half - we want to collect and address all feedback on FMV from partners and compilers communities.

**Pull request merged**